pkgs/top-level/lua-packages.nix

```diff
@@ -33,12 +33,19 @@ let
       sha256 = "16fffbrgfcw40kskh2bn9q7m3gajffwd2f35rafynlnd7llwj1qj";
     };
 
+    patchPhase = ''
+      sed -e "s,MACOSX_DEPLOYMENT_TARGET=10.4,MACOSX_DEPLOYMENT_TARGET=10.10," -i Makefile
```

Comment:
I'd suggest just removing the `MACOSX_DEPLOYMENT_TARGET` so `make` inherits the one we set in the stdenv.